### ISSUE TITLE
[Bug] Drafts cannot be deleted, only archived

### ISSUE THREAD

**Issue by tdnzr** [ForumMagnum/ForumMagnum#7797]
In one's user page, one can click a small icon next to a post to "Archive this draft". This draft can then be displayed again by clicking on the small gear icon and activating "Show Archived".

Separately, drafts (but not posts) can ostensibly be **deleted**, via context menu -> "Delete Post". (Side note: if only drafts can be deleted, this option could probably be renamed to "Delete Draft"). Choosing this option opens a popup which says "Are you sure you want to delete this post?". (Again, this should maybe say "delete this draft") Doing this makes it look like the draft is deleted, but in fact it's still visible once you choose "Show Archived" again.

In summary, (my / some / all) Drafts currently cannot be deleted, only archived.

(Reported on LW; I don't know if the same problem occurs on the EA Forum.)
